packages/js/src/first-time-configuration/tailwind-components/unsaved-changes-modal.js

```diff
@@ -44,18 +44,17 @@ export default function UnsavedChangesModal( { hasUnsavedChanges, title, descrip
 	 * @returns {void}
 	 */
 	const popStateEventHandler = useCallback( () => {
-		if ( hasUnsavedChanges ) {
+		if ( hasUnsavedChanges && location.href.indexOf( "#top" ) === -1 ) {
```

Comment:
I think we should make the IF stricter. Maybe by checking for the existence of `wpseo_dashboard` as well?